src/lib/hooks/useNetworkFees.ts

```diff
@@ -0,0 +1,198 @@
+    };
+
+    const getWallet = useCallback(
+        async (coin: string, network: string) =>
+            profile.walletFactory().generate({ coin, network }),
+        [profile],
+    );
+
+    const createStubTransaction = useCallback(
+        async ({ coin, type, getData, stub }: CreateStubTransactionProperties) => {
+            const { mnemonic, wallet } = await getWallet(
+                coin.network().coin(),
+                coin.network().id(),
+            );
+
+            const signatory = stub
+                ? await wallet.signatory().stub(mnemonic)
+                : await wallet.signatory().mnemonic(mnemonic);
+
+            return (coin.transaction() as any)[type]({
+                data: getData(wallet),
+                nonce: '1',
+                signatory,
+            });
+        },
+        [getWallet],
+    );
+
+    const calculateBySize = useCallback(
+        async ({ coin, data, type }: CalculateBySizeProperties): Promise<TransactionFees> => {
+            try {
+                const transaction = await createStubTransaction({
+                    coin,
+                    getData: (senderWallet) => {
+                        if (type === 'multiSignature') {
+                            return getMuSigData(senderWallet, data);
+                        }
+
+                        return data;
+                    },
+                    stub: type === 'multiSignature',
+                    type,
+                });
+
+                const [min, avg, max] = await Promise.all([
+                    coin.fee().calculate(transaction, { priority: 'slow' }),
+                    coin.fee().calculate(transaction, { priority: 'average' }),
+                    coin.fee().calculate(transaction, { priority: 'fast' }),
+                ]);
+
+                return {
+                    avg: roundAndFormat(avg),
+                    max: roundAndFormat(max),
+                    min: roundAndFormat(min),
+                    static: roundAndFormat(min),
+                };
+            } catch {
+                return {
+                    avg: '0',
+                    max: '0',
+                    min: '0',
+                    static: '0',
+                };
+            }
+        },
+        [createStubTransaction],
+    );
+
+    const calculate = useCallback(
+        async ({ coin, network, type, data }: CalculateProperties): Promise<TransactionFees> => {
+            let transactionFees: Services.TransactionFee;
+
+            const coinInstance = profile.coins().get(coin, network);
+
+            try {
+                transactionFees = env.fees().findByType(coin, network, type);
+            } catch {
+                await env.fees().syncAll(profile);
+
+                transactionFees = env.fees().findByType(coin, network, type);
+            }
+
+            if (
+                !!data &&
+                (coinInstance.network().feeType() === 'size' || type === 'multiSignature')
```

Comment:
pretty much everything related to musig can go, likewise for the `size` feetype as that's not used on ark at this time either